### PR TITLE
Add Desktop to file dialogue as drive in unix

### PR DESCRIPTION
**Pull request by ShlomiRex**
When I click "Import" (or Create Project) in Project Manager and want to select desktop from the drives on top-left, I can't, it's not there. So I added it.
It should be added to any dialogue that uses the File Dialogue
It's a small commit. It's my first pull ever / first contribution. Please accept :P

### CODE REVIEW

**Comment by groud** [comment]
The `$HOME/Desktop` isn't used by all linux distributions. I use Archlinux and don't use a Desktop.
So I'd rather avoid adding a wrong path into the list on drives (If we can consider this a drive).

**Comment by Calinou** [comment]
The right way to do so is to read the value of the `XDG_DESKTOP_DIR` environment variable, and default to `$HOME/Desktop` if it's not set: https://wiki.archlinux.org/index.php/XDG_user_directories

**Comment by ShlomiRex** [comment]
> The right way to do so is to read the value of the `XDG_DESKTOP_DIR` environment variable, and default to `$HOME/Desktop` if it's not set: https://wiki.archlinux.org/index.php/XDG_user_directories

Alright I'll fix it.
Also, why does it failed a check? What should I do? It failed and says: Static checks (clang-format) + Documentation checks

**Comment by ShlomiRex** [comment]
> The `$HOME/Desktop` isn't used by all linux distributions. I use Archlinux and don't use a Desktop.
> So I'd rather avoid adding a wrong path into the list on drives (If we can consider this a drive).

Thanks Ill fix it

**Comment by Calinou** [comment]
> Also, why does it failed a check? What should I do? It failed and says: Static checks (clang-format) + Documentation checks

You need to fix style issues reported by clang-format. Check the full Travis CI log by clicking on the red "X" icon then click **Details** and choose the Static checks job.

We recommend installing clang-format locally so you can do this automatically. There are IDE plugins available to integrate clang-format, but you can also install the Git pre-commit hook by running this command in Godot's root directory:

```bash
cp misc/hooks/* .git/hooks
```

**Comment by ShlomiRex** [comment]
> The right way to do this is to read the value of the `XDG_DESKTOP_DIR` environment variable, and default to `$HOME/Desktop` if it's not set: https://wiki.archlinux.org/index.php/XDG_user_directories
> 
> The OS singleton has a `get_environment()` method you can use for this.

How do I convert the String object of "$HOME/Desktop" to be a String that is equal to "/home/shlomi/Desktop/" in my case?
I tried what you told me, I included "drivers/unix/os_unix.h" but i don't know how to use the "get_environment()" method, what it does? I give it a String object that is equal to "$HOME/Desktop" but it gives me errors:

`drivers/unix/dir_access_unix.cpp:266:51: error: cannot call member function 'virtual String OS_Unix::get_environment(const String&) const' without object
  266 |       String test = OS_Unix::get_environment(fpath);`

**Comment by Calinou** [comment]
@ShlomiRex Indeed, `OS::get_singleton()->get_system_dir()` with the desktop constant is the way to go. No need to handle environment variables ourselves.

**Comment by ShlomiRex** [comment]
> @ShlomiRex Indeed, `OS::get_singleton()->get_system_dir()` with the desktop constant is the way to go. No need to handle environment variables ourselves in this case.
> 
> Apologies for the confusion, I forgot about that method.

Alright now I need to squash the commits into 1?

**Comment by Calinou** [comment]
@ShlomiRex Yes, see [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html) for details.

File: drivers/unix/dir_access_unix.cpp [hunk 1]
```diff
@@ -227,17 +229,17 @@ static void _get_drives(List<String> *list) {
 			list->push_back(home);
 		}
 
-		// Check $HOME/.config/gtk-3.0/bookmarks
 		char path[1024];
 		snprintf(path, 1024, "%s/.config/gtk-3.0/bookmarks", home);
 		FILE *fd = fopen(path, "r");
+		String fpath;
```
Comment (aaronfranke) [inline drivers/unix/dir_access_unix.cpp:235]: Why declare `fpath` outside of the while loop? There isn't really a purpose to have it exist in this scope, the value isn't re-used across loops, and the additional usage that this PR adds is completely unrelated.

You should change this back to the way it was, and for the new behavior, change `fpath =` to `String desktop_path =` etc.

**Review by aaronfranke — DISMISSED** [review]

File: drivers/unix/dir_access_unix.cpp (hunk 1 above)
Comment (akien-mga) [reply drivers/unix/dir_access_unix.cpp]: Fixed.

**Comment by akien-mga** [comment]
Sorry for the delay further reviewing this. I force pushed a rebase to address the review comment and re-add a wrongly removed comment (and I improved comment style a bit while at it).

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:
